core/templates/services/question-validation.service.ts

```diff
@@ -94,6 +112,17 @@ export class QuestionValidationService {
     if (!atLeastOneAnswerCorrect) {
       return 'At least one answer should be marked correct';
     }
+    if (
+      question._stateData.interaction.answerGroups.length > 1 ||
+      question._stateData.interaction.answerGroups[0].rules.length > 1
+    ) {
+      if (!this.areAnswerGroupsNotEqual(question)) {
+        return (
+          'Please ensure learner answer 1 in Oppia response 2' +
```

Comment:
Why this specific check? Shouldn't it be generalized to any rule in any answer group? (and we say that 'rule X in answer group Y is a duplicate of rule W in answer group Z)